### DISCUSSION TITLE
Problem with jfree.ui

### DISCUSSION THREAD

**Discussion by gw1500se1** [jfree/jfreechart#384] [General]
I moved to a new Windows platform so I had to install Eclipse (2023-09) an external archives (jfreechart-1.5.4 and jcommon-1.0.8) which are newer than what I was using. After cloning my app I am getting the following errors:

`chart.setInsets(new RectangleInsets(4,20,4,20));`
 The method setInsets(org.jfree.chart.ui.RectangleInsets) in the type Plot is not applicable for the arguments (org.jfree.ui.RectangleInsets)

`label.setRotationAnchor(TextAnchor.BASELINE_CENTER);`
The method setRotationAnchor(org.jfree.chart.ui.TextAnchor) in the type XYTextAnnotation is not applicable for the arguments (org.jfree.ui.TextAnchor)

`label.setTextAnchor(TextAnchor.BASELINE_CENTER);`
The method setTextAnchor(org.jfree.chart.ui.TextAnchor) in the type XYTextAnnotation is not applicable for the arguments (org.jfree.ui.TextAnchor)

I'm guessing that my code has been deprecated in the new version but I cannot find any information on migration. Can someone help me correct these statements to conform to the new version? TIA.

**Comment by trashgod** [comment]
As noted in [Migration](https://github.com/jfree/jfreechart#migration), `JCommon` has been integrated with `JFreeChart`.

**Reply by gw1500se1** [reply]
Thanks for the link. I removed jcommon from the build path and added some imports. Problem resolved.
